### PR TITLE
Request less data when making history api calls where possible

### PR DESCRIPTION
**Pull request by bdraco**
## Proposed change

Supports https://github.com/home-assistant/core/pull/35822

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by bramkragten** [comment]
I don't think this will work in all cases, for a climate device we show both the set temperature, the temperature and if it was heating and cooling in the history graph.

image

https://github.com/home-assistant/frontend/blob/dev/src/components/state-history-chart-line.js#L153-L155

So we should make `minimal_response` optional for `fetchRecent` and check if the values are there in the processing.

After that we have to identify what users of `fetchRecent` do need the attributes of every state change and make changes.

**Comment by bdraco** [comment]
Right, for the domains that need attributes we still send everything:

https://github.com/home-assistant/core/pull/35822/files#diff-186d4b915b87ade6b9adea5e2a4bd65dR48

We could optimize it a bit more but I stopped there because it was diminishing returns.

**Comment by bdraco** [comment]
Backend changes have merged

File: src/data/history.ts [hunk 1]
```diff
@@ -153,6 +162,9 @@ const processLineChartEntities = (
 
       if (
         processedStates.length > 1 &&
+        processedState.attributes &&
+        processedStates[processedStates.length - 1].attributes &&
+        processedStates[processedStates.length - 2].attributes &&
```
Comment (bramkragten) [inline src/data/history.ts:167]: If we don't have attributes on all we should still check if the states are the same to not push the same point to the chart?

Comment (bramkragten) [reply src/data/history.ts:167]: So I think we should change the `equalState` function instead of only calling that when we have attributes

Comment (bdraco) [reply src/data/history.ts:167]: Done, and reverted this change

File: src/data/history.ts [hunk 2]
```diff
@@ -92,6 +92,7 @@ const equalState = (obj1: LineChartState, obj2: LineChartState) =>
   obj1.state === obj2.state &&
   // They either both have an attributes object or not
```
[the remark this answers is not in the document]
Comment (bdraco) [reply src/data/history.ts]: Adjusted!

**Review by bramkragten — APPROVED** [review]

**Comment by bdraco** [comment]
Thanks @bramkragten

**Pull request merged**